### PR TITLE
[RFR] Display error message when the form doesn't validate

### LINKED ISSUE #1513: Display error message when the form doesn't validate
When I click the save button and the form doesn't validate, the errors may not be on screen, so I sometimes wonder why the action didn't succeed. 

We should display a generic message, like "Save failed: validation error", maybe in a Snackbar?

### PR DESCRIPTION
**Pull request by djhi**
Fix #1513

screen

### CODE REVIEW

**Comment by fzaninotto** [comment]
You should use a `warning` type in the notification IMO.

File: packages/ra-language-french/index.js [hunk 1]
```diff
@@ -50,6 +50,7 @@ module.exports = {
                 "L'URL saisie est incorrecte, ou vous avez suivi un mauvais lien.",
             loading:
                 'La page est en cours de chargement, merci de bien vouloir patienter.',
+            invalid_form: `Le formulaire n'est pas valide. Veuillez vérifier les champs en erreur.`,
```
Comment (fzaninotto) [inline packages/ra-language-french/index.js]: Isn't it too long?

File: packages/react-admin/src/mui/button/SaveButton.spec.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { SaveButton } from './SaveButton';
 
 const translate = label => label;
 
-describe('<SaveButton />', () => {
+describe.only('<SaveButton />', () => {
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/button/SaveButton.spec.js]: nooooooooo...

Don't you use Jest?

File: packages/react-admin/src/mui/button/SaveButton.js [hunk 3]
```diff
@@ -37,12 +38,22 @@ const sanitizeRestProps = ({
...
+            redirect,
+            saving,
+            showNotification,
+        } = this.props;
+
+        if (saving) {
             // prevent double submission
             e.preventDefault();
         } else {
+            if (invalid) {
```
Comment (fzaninotto) [inline packages/react-admin/src/mui/button/SaveButton.js]: I'm not sure what `invalid` means for redux-form, but can you confirm this notification shows only *after* the user has submitted the form?

File: packages/react-admin/src/mui/button/SaveButton.spec.js (hunk 2 above)
Comment (djhi) [reply packages/react-admin/src/mui/button/SaveButton.spec.js]: oups ^^

File: packages/react-admin/src/mui/button/SaveButton.js (hunk 3 above)
Comment (djhi) [reply packages/react-admin/src/mui/button/SaveButton.js]: I confirm because we only check for an invalid form inside the click handler :)

**Pull request merged**